src/main/java/org/webjars/WebJarExtractor.java

```diff
@@ -97,227 +91,87 @@ public void extractAllBowerComponentsTo(File to) throws IOException {
-                } catch (URISyntaxException e) {
-                    file = new File(url.getPath());
+                    moduleId[0] = getJsonModuleId(new String(byteArray));
+                } catch (IOException e) {
+                    log.error("Could not get moduleId", e);
                 }
-                log.debug("Found file system webjar: {}", file);
-                File[] webjars;
-                if (name == null) {
-                    webjars = file.listFiles();
-                } else {
-                    webjars = new File[] {file};
-                }
-                if (webjars != null) {
-                    for (File webjar: webjars) {
-                        if (webjar.isDirectory()) {
-                            File[] versions = webjar.listFiles();
-                            if (versions != null) {
-                                for (File version: versions) {
-                                    if (version.isDirectory()) {
-                                        String moduleId;
-                                        if (useModuleName) {
-                                            moduleId = getFileNodeModuleIdEntry(new File(version, moduleNameFile));
-                                        } else {
-                                            moduleId = webjar.getName();
-                                        }
-                                        if (moduleId != null) {
-                                            File copyTo = new File(to, moduleId);
-                                            copyDirectory(version, copyTo, webjar.getName());
-                                        }
-
-                                    } else {
-                                        log.debug("Filesystem webjar version {} is not a directory", version);
-                                    }
-                                }
-                            } else {
-                                log.debug("Filesystem webjar has no versions: {}", webjar);
-                            }
-                        } else {
-                            log.debug("Filesystem webjar {} is not a directory", webjar);
+            }
+        });
+
+        return moduleId[0];
+    }
+
+    private void extractResourcesTo(final String webJarName, final String webJarVersion, final String moduleNameFile, final ResourceList webJarResources, final File to) {
+        final boolean useModuleName = moduleNameFile != null;
+
+        final String webJarId = useModuleName ? getModuleId(webJarResources, moduleNameFile) : webJarName;
+
+        webJarResources.forEachInputStream(new ResourceList.InputStreamConsumer() {
+            @Override
+            public void accept(Resource resource, InputStream inputStream) {
+                final String prefix = WEBJARS_PATH_PREFIX + File.separator + webJarName + File.separator + (webJarVersion == null ? "" : webJarVersion + File.separator);
```

Comment:
Using `File.separator` here is incorrect. In Windows backslash will be used, however the defined path separator in JAR files is `/` - it is not platform-dependent. Fix is here: #119